src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java

```diff
@@ -648,16 +654,30 @@ private String buildSearchTransactionsQuery(final TransactionsSearchDto transact
 
 			if (transactionsSearchDto.getTransactionType() != null) {
 				query.append("and (CASE"
-					+ "        WHEN trnstat = " + TransactionStatus.COMMITTED.getIntValue() + " AND trnqty >= 0 THEN '"
+					+ "        WHEN trntype = " + TransactionType.DEPOSIT.getId() + " THEN '"
```

Comment:
transactionType in transactionsSearchDto should become an Integer (or List of Integer if we want to be more flexible) so this fragment of query will be simpler

Reply:
Done